source/slang/include-file-system.h

```diff
@@ -0,0 +1,81 @@
+#ifndef SLANG_INCLUDE_FILE_SYSTEM_H_INCLUDED
+#define SLANG_INCLUDE_FILE_SYSTEM_H_INCLUDED
+
+#include "../../slang.h"
+#include "../../slang-com-helper.h"
+#include "../../slang-com-ptr.h"
+
+namespace Slang
+{
+
+class IncludeFileSystem : public ISlangFileSystemExt
```

Comment:
Should this be `DefaultFileSystem`, since it isn't really just for `#include` handling. Ideally *all* file loading should go through this layer eventually.

Reply:
It could be DefaultFileSystem, and that'd be fine. 
I'm not sure if here if you mean in terms of replacing *all*,  replacing 'IncludeHandler'. But as it stands that has yet more semantics - in that it tracks files that are referenced, and knows the search paths. 
That could be done and how it's done is a little odd (with virtual interface) - I guess so preprocessor doesn't have to have any extra knowledge.

Reply:
I just meant the rename in this case, and nothing about tangling this up with the responsibilities of the `IncludeHandler` type. Rather, my point was that the type in this file is really the `DefaultFileSystem` (no more, no less), and shouldn't have `Include` in its name, because that should be handled at another layer.

I'm not sure who precisely should own the list of include paths (should it just live on the `CompileRequest` directly, with no intermediate objects?).